teku/src/main/java/tech/pegasys/teku/config/TekuConfiguration.java

```diff
@@ -47,17 +56,29 @@ public BeaconChainConfiguration beaconChain() {
     return beaconChainConfig;
   }
 
+  public ValidatorClientConfiguration validatorClient() {
+    return validatorClientConfig;
+  }
+
+  public void validate() {
+    globalConfiguration.validate();
+    validatorConfig.validate();
```

Comment:
If possible, I'd suggest validating and throwing an `IllegalArgumentException` when building the `validatorConfig`, so that we don't have to explicitly invoke this `validate()` method.  For example, `WeakSubjectivityConfig` errors are handled when `TekuConfiguration` builds [here](https://github.com/PegaSysEng/teku/blob/master/teku/src/main/java/tech/pegasys/teku/cli/BeaconNodeCommand.java#L328-L339).  This prevents us from creating invalid configs in the first place.

Reply:
Good idea, done. Cleaned up a bunch of confusion between `null` and empty lists at the same time.